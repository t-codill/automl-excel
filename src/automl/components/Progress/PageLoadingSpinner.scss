.page-loading-spinner {
    height: 100%;
    padding: 2em;
    box-sizing: border-box;
}