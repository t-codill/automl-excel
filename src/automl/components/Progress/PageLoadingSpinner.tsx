import { Spinner, SpinnerSize } from "office-ui-fabric-react";
import * as React from "react";

import "./PageLoadingSpinner.scss";

const pageLoadingSpinner: React.FunctionComponent<{ label?: string }> = (props) => {
    return <Spinner className="page-loading-spinner" size={SpinnerSize.large} label={props.label || "Loading..."} />;
};

export { pageLoadingSpinner as PageLoadingSpinner };
