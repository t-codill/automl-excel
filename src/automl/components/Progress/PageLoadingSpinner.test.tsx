import { shallow } from "enzyme";
import * as React from "react";
import { PageLoadingSpinner } from "./PageLoadingSpinner";

describe("PageLoadingSpinner", () => {
    it("should render", () => {
        const wrapper = shallow(<PageLoadingSpinner />);
        expect(wrapper)
            .toMatchSnapshot();
    });
    it("should render with label", () => {
        const wrapper = shallow(<PageLoadingSpinner label={"label1"} />);
        expect(wrapper)
            .toMatchSnapshot();
    });
});
