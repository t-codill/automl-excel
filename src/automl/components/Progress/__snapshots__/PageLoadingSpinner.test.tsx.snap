// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLoadingSpinner should render 1`] = `
<StyledSpinnerBase
  className="page-loading-spinner"
  label="Loading..."
  size={3}
/>
`;

exports[`PageLoadingSpinner should render with label 1`] = `
<StyledSpinnerBase
  className="page-loading-spinner"
  label="label1"
  size={3}
/>
`;
